packages/amplify-console-hosting/src/utils/build-utils.js

```diff
@@ -37,32 +38,28 @@ function zipFile(sourceDir, destFilePath, extraFiles) {
   });
 }
 
-function run(command, projectDirectory) {
+async function run(command, projectDirectory) {
   if (!command) {
-    throw new Error('Missing build command');
-  }
-
-  return new Promise((resolve, reject) => {
-    const execution = executeCommand(command, { cwd: projectDirectory, env: process.env, stdio: 'inherit' });
-
-    let rejectFlag = false;
-    void execution.on('exit', (code) => {
-      if (code === 0) {
-        resolve();
-      } else if (!rejectFlag) {
-        rejectFlag = true;
-        reject(code);
-      }
+    throw new AmplifyError('ProjectBuildCommandError', {
+      message: `Missing build command: ${command}`,
+      resolution: 'Run `amplify configure project` to set the build command.',
+      link: 'https://docs.amplify.aws/cli/start/workflows/#amplify-configure-project',
     });
+  }
 
-    void execution.on('error', (err) => {
-      console.log(chalk.red('command execution terminated with error'));
-      if (!rejectFlag) {
-        rejectFlag = true;
-        reject(err);
-      }
-    });
-  });
+  try {
+    await executeCommand(command, { cwd: projectDirectory, env: process.env, stdio: 'inherit' });
+  } catch (cause) {
+    throw new AmplifyError(
+      `ProjectBuildCommandError`,
+      {
+        message: `Failed to run frontend build command: ${command}`,
```

Comment:
Seems more accurate to say `Build command ${command} exited with failure`

Reply:
that sounds good to me! updated